packages/ensnode-sdk/src/client.ts

```diff
@@ -0,0 +1,231 @@
+import type { Address } from "viem";
+import type { CoinType, Name } from "./utils/index.js";
+
+/**
+ * Default ENSNode API endpoint URL
+ */
+export const DEFAULT_ENSNODE_API_URL = "https://api.alpha.ensnode.io" as const;
+
+/**
+ * Configuration options for ENSNode API client
+ */
+export interface ClientOptions {
+  /** The ENSNode API endpoint URL */
+  endpointUrl: URL;
+  /** Whether to enable debug tracing */
+  debug?: boolean;
+}
+
+/**
+ * Selection criteria for forward resolution (name to records)
+ */
+export interface ForwardResolutionSelection {
+  /** Array of coin types to resolve addresses for */
+  addresses?: CoinType[];
+  /** Array of text record keys to resolve */
+  texts?: string[];
+}
+
+/**
+ * Resolved records response
+ */
+export interface Records {
+  /** The canonical name (only available in reverse resolution) */
+  name?: string;
+  /** Resolved addresses by coin type */
+  addresses?: Record<string, string>;
+  /** Resolved text records */
+  texts?: Record<string, string>;
+}
+
+/**
+ * Base API response structure
+ */
+export interface BaseResponse {
+  /** Resolved records */
+  records: Records;
+  /** Debug trace information (if debug enabled) */
+  trace?: any;
+}
+
+/**
+ * Forward resolution response (name to records)
+ */
+export interface ForwardResponse extends BaseResponse {}
+
+/**
+ * Reverse resolution response (address to name)
+ */
+export interface ReverseResponse extends BaseResponse {}
+
+/**
+ * API error response
+ */
+export interface ErrorResponse {
+  error: string;
+  code?: string;
+  details?: Record<string, unknown>;
+}
+
+/**
+ * Complete ENSNode API client interface
+ */
+export interface Client {
+  /**
+   * Resolve an ENS name to records (forward resolution)
+   */
+  resolveName(name: Name, selection?: ForwardResolutionSelection): Promise<ForwardResponse>;
+
+  /**
+   * Resolve an address to its primary name (reverse resolution)
+   */
+  resolveAddress(address: Address, chainId?: number): Promise<ReverseResponse>;
+
+  /**
+   * Get the current client options
+   */
+  getOptions(): Readonly<ClientOptions>;
+}
+
+/**
+ * Complete ENSNode API client
+ *
+ * Provides access to all ENSNode API functionality through a unified interface.
+ * Supports resolution, configuration, and indexing status operations.
+ *
+ * @example
+ * ```typescript
+ * // Create client with default options
+ * const client = new ENSNodeClient();
+ *
+ * // Use resolution methods
+ * const nameResult = await client.resolveName("vitalik.eth", {
+ *   addresses: [60],
+ *   texts: ["avatar"]
+ * });
+ *
+
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // Custom configuration
+ * const client = new ENSNodeClient({
+ *   endpointUrl: new URL("https://custom-api.ensnode.io"),
+ *   debug: true
+ * });
+ * ```
+ */
+export class ENSNodeClient implements Client {
```

Comment:
`debug` config is on a per-request basis, it may or may not be supported across all of the ENSNode apis, it just happens to be so on all of the resolution apis. understandable assumption, apologies.